src/app/ui-kit/button/button.component.ts

```diff
@@ -0,0 +1,23 @@
+import { Component, Input, OnInit } from '@angular/core';
+
+@Component({
+  selector: 'app-button',
+  templateUrl: './button.component.html',
+})
+export class ButtonComponent implements OnInit {
+  @Input() type: 'accent' | 'ghost' = 'accent';
+  @Input() size: 'sm' | 'md' = 'sm';
+  @Input() iconState: string = '';
+  @Input() disabled = false;
+
+  classes: string = '';
+  imageSrc: string = '../../..//assets/images/chevron.svg';
+  iconDeg: string = '';
+  iconStyles: any = '';
+
+  ngOnInit(): void {
```

Comment:
ngOnInit works only once, this component should read values multiple times. You can use a getter: get classes(): string {...}